dbms/src/Columns/ColumnString.cpp

```diff
@@ -747,20 +1013,45 @@ void ColumnString::deserializeAndInsertFromPosForColumnArray(
 
     size_t size = pos.size();
     size_t char_size = chars.size();
-    for (size_t i = 0; i < size; ++i)
+    if constexpr (add_terminating_zero)
```

Comment:
It seems we can also move this branch inside the for-loop to reuse more common code.

Reply:
save reason as L798

Reply:
save reason as L798

Reply:
ok